core/src/martian/interceptors.cljc

```diff
@@ -83,22 +84,34 @@
               (update ctx ::tc/queue #(into (into tc/queue i) %))
               ctx))})
 
+(defn- prepare-multipart-request-headers [request]
+  (update request :headers dissoc "Content-Type" "content-type" :content-type))
+
 (defn encode-request [encoders]
   {:name ::encode-request
    :encodes (keys encoders)
    :enter (fn [{:keys [request handler] :as ctx}]
             (let [has-body? (:body request)
-                  content-type (and has-body?
-                                    (not (get-in request [:headers "Content-Type"]))
-                                    (encoding/choose-content-type encoders (:consumes handler)))
-                  multipart? (= "multipart/form-data" content-type)
-                  {:keys [encode]} (encoding/find-encoder encoders content-type)]
-              (cond-> ctx
-                      has-body? (update-in [:request :body] encode)
-                      ;; NB: Luckily, all target HTTP clients — clj-http (but not lite), http-kit,
-                      ;;     hato, and even babashka/http-client — all support the same syntax.
-                      multipart? (update :request set/rename-keys {:body :multipart})
-                      content-type (assoc-in [:request :headers "Content-Type"] content-type))))})
+                  content-type (when (and has-body?
+                                          (not (get-in request [:headers "Content-Type"])))
+                                 (encoding/choose-content-type encoders (:consumes handler)))
+                  ;; NB: There are many possible subtypes of multipart requests.
+                  multipart? (when content-type (str/starts-with? content-type "multipart/"))
+                  {:keys [encode]} (encoding/find-encoder encoders content-type)
+                  encoded-request (cond-> request
+
+                                          has-body?
+                                          (update :body encode)
+
+                                          multipart?
+                                          ;; NB: Luckily, all target HTTP clients — clj-http (but not lite), http-kit,
+                                          ;;     even hato and org.babashka/http-client — all support the same syntax.
+                                          (-> (set/rename-keys {:body :multipart})
+                                              (prepare-multipart-request-headers))
+
+                                          (and content-type (not multipart?))
+                                          (assoc-in [:headers "Content-Type"] content-type))]
+              (assoc ctx :request encoded-request)))})
```

Comment:
This is one of the primary changes. Removes an existing "Content-Type" request header, if any, from _all_ multipart requests.